### DISCUSSION TITLE
Modulo kata 6

### DISCUSSION THREAD

**Discussion by MauricioNovelo** [LaunchX-InnovaccionVirtual/CursoIntroPython#26] [Ideas]
Tengo la duda de como lograr no repetir la misma linea de código en mis ejercicios
![image](https://user-images.githubusercontent.com/83683322/153096394-7b2cf8fb-b807-411f-94f4-95bd1daeab56.png)
Existe alguna función o manera de lograrlo?
Igual en este código se soluciona el error de que índex no encuentra el valor en lista.

**Comment by jstoledano** [comment]
No entiendo qué dices que se repite. Pero se repite porque lo estás escribiendo dos veces. 

Pero ahí te van unos consejos y observaciones.

1. Declaras la variable `planeta` y está bien, porque la vas a ocupar, aprovecha para convertir tu entrada y te ahorras una variable.
2. Luego le poner la mayúscula inicial para poder compararla con los elementos de tu lista. Es mejor usar `upper()` o `lower()` en ambos elementos porque con tu método `tierra` se vuelve `Tierra` y checa con tu lista, pero `tieRra` se vuelve `TieRra` y ya no checa. Esto es mas engorroso porque también hay que trabajar con los elementos de tu lista, pero piénsalo para la próxima.
3. No sé porque nos recomiendan usar `index`, pero en este caso no es necesario.  Este código también funciona y no se usa `index`:
 
```python
planetas = ['Mercurio', 'Venus', 'Tierra', 'Marte', 'Júpiter', 'Saturno', 'Urano', 'Neptuno']
planeta = input("Escribe el nombre de un planeta").title()
if planeta in planetas:
    print(f'El planeta {planeta} sí está en el sistema solar.')
else:
    print(f'El planeta {planeta} no está en el sistema solar.')
```

**Reply by MauricioNovelo** [reply]
Si ya cheque con la variable TierRa y siempre hace la función, el método title() deja bien las variables en uso para que se puedan checar en la lista, tengo echo el código pensando que el que escriba el nombre del planeta pueda volver a escribir un planeta valido, por eso esta el while

**Reply by PlayerOneV** [reply]
Buenas noches, me podrías decir como agregaste el bloque de cogido Python aqui en github por favor.

**Reply by MauricioNovelo** [reply]
Ahi donde dice codigo y markdown, le das a codigo para agregar bloques python y el markdown es como para escribir, igual le puedes dar formatos interesante.
Ya cheque lo del ciclo while True y queda mas corto, pero con un poco mas de logica

**Reply by Oscar19260** [reply]
> Si ya cheque con la variable TierRa y siempre hace la función, el método title() deja bien las variables en uso para que se puedan checar en la lista, tengo echo el código pensando que el que escriba el nombre del planeta pueda volver a escribir un planeta valido, por eso esta el while

En todo caso tendrías que usar un while true, para que el código este siempre activo y puedas seguir introduciendo nombres y usar una condición para que lo pueda finalizar con un break.

**Reply by Dfredude** [reply]
> En todo caso tendrías que usar un while true, para que el código este siempre activo y puedas seguir introduciendo nombres y usar una condición para que lo pueda finalizar con un break.
De hecho el while loop esta bien hecho, usar break es opcional realmente.

**Comment by PlayerOneV** [comment]
Le estas pidiendo al usuario que introduzca el nombre de un planeta dos veces, si tu intención era preguntarle hasta que ingrese un planeta que se encontrara en el sistema solar podías usar un ciclo "do while".
while True:
  //tu codigo
  if falla_condición:
    break //Sales del bucle while

**Reply by MauricioNovelo** [reply]
Nop, checa bien el código, si escribes bien el nombre del planeta ya no te vuelve a pedir otra vez, a diferencia que lo escribas mal, ahí si te volvería a pedir un planeta valido que se encuentre en la lista. Y la parte 
planetas_index = planetas.index(planetatitle) if planetatitle in planetas else -1
es solo para que no arroje el error de valor no esta en lista.
Sip, ya cheque el while true, veré si se puede acortar con esto
![image](https://user-images.githubusercontent.com/83683322/153114244-5a0c03a8-1982-4249-b342-d411035cb4bb.png)
así queda

**Comment by Dfredude** [comment]
Me gusto la forma en que desarrollaste el programa, no habia pensado en hacerlo robusto para solo aceptar una entrada valida. Creo que es una buena idea, aunque creo que nadie entendio realmente lo que querias hacer. Este **_script_** me gusto para reducir repeticion de codigo de tu programa, y funciona exactamente igual. 
```python
planet_index = -1
while planet_index == -1:
    planet = str(input("Enter a planet to search for in the database (Capitalized): "))
    planet_index = planets.index(planet) if planet in planets else -1
    if planet_index == -1: print("Please enter a valid planet within our solar system.")
print(f"We found {planet}!")
```

Por cierto por ahí te aconsejaron no usar `index()`, pero **_tenemos que usar_** `index()` para terminar la otra parte del ejercicio, solo queria remarcar eso. Espero te haya ayudado.

**Comment by CarlosCorales** [comment]
Yo tengo una duda, el comando .append no me deja usarlo, sabran por que?

**Reply by jstoledano** [reply]
Porque el método `.append()` solo funciona en listas y tal vez lo estés usando con una tupla u otro objeto, mira:

```
>>> tupla = ()
>>> type(tupla)
<class 'tuple'>
>>> tupla.append=(1,2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'tuple' object has no attribute 'append'
>>> lista = []
>>> type(lista)
<class 'list'>
>>> lista.append(1)
>>> lista.append(2)
>>> lista
[1, 2]
```
